rcl/src/rcl/node.c

```diff
@@ -408,6 +411,9 @@ rcl_node_init(
   if (NULL != remapped_node_name) {
     allocator->deallocate(remapped_node_name, allocator->state);
   }
+  if (!node_secure_root) {
```

Comment:
Is this intended to be `node_secure_root` or `NULL != node_secure_root`?

Reply:
I think if `node_secure_root` is NULL, it will evalute to false, and thus will only deallocate if it's not NULL.

Reply:
I think you might be overlooking the `!`

Reply:
If it's OK to `deallocate` a null pointer, we could do without this. That'd be nice if it were the case because it would nicely align with the behavior of `free`.

Reply:
@sloretz @thomas-moulard going to deallocate without checking the value here.